connect/clients/ipfs.py

```diff
@@ -0,0 +1,124 @@
+"""
+ipfs.py
+
+IPFS client services that support REST calls for persistence of records with an
+IPFS cluster
+"""
+import logging
+import httpx
+import json
+from connect.config import get_settings
+
+
+logger = logging.getLogger(__name__)
+_ipfs_client = None
+
+
+# ******************************************
+# IPFS client services for persistence
+# ******************************************
+class IPFSClient:
+    def __init__(self, ipfs_cluster_uri, ipfs_cluster_replication_factor):
+        self._ipfs_cluster_uri = ipfs_cluster_uri
+        self._ipfs_cluster_replication_factor = ipfs_cluster_replication_factor
+        self._add_to_ipfs_cluster_uri = self._ipfs_cluster_uri + '/add'
+        self._get_ipfs_cluster_peers_uri = self._ipfs_cluster_uri + '/peers'
+        self._unpin_cid_ipfs_cluster_uri = self._ipfs_cluster_uri + '/pins/'
+
+    async def persist_json_to_ipfs(self, payload_dict):
+        """
+        Allows persistence of JSON payloads (as a dict) into an IPFS cluster
+        The IPFS cluster instance being connected to is configurable.
+
+        Content is pinned by default during the add operation.
+
+        :param payload_dict
+        :return content_cid: A unique ID addressing the content added to the
+                             IPFS cluster. Returns None if a non 200 response
+                             code is returned from the IPFS Cluster.
+        """
+        if not payload_dict:
+            error_msg = 'dict payload not provided for persistence into IPFS'
+            logger.error(error_msg)
+            raise ValueError(error_msg)
+
+        async with httpx.AsyncClient() as client:
+            with open('payload.json', 'w') as payload_temp_file:
+                json.dump(payload_dict, payload_temp_file)
+
+            files = {'upload-file': ('payload.json', open('payload.json', 'r'),
+                     'multipart/form-data')}
+            response = await client.post(self._add_to_ipfs_cluster_uri,
+                                         files=files)
+
+            if response.status_code == 200:
+                response_dict = json.loads(response.json())
+                info_msg = "Successfully added and pinned to IPFS -- " + \
```

Comment:
This can probably be a single line

```
logger.info(f"Successfully added/pinned to IPFS -- cid: {response_dict['cid']}")
```
You can see what makes the most sense after the black formatter runs.

Reply:
I will defer to the formatter